pkg/testing/utils/utils.go

```diff
@@ -154,6 +154,35 @@ func BuildProc(comm string, uid, gid uint32) Process {
 	}
 }
 
+// NormalizeParentTid normalizes the parent TID to the first thread (main thread)
+// of the process by reading /proc/$tid/task/ directory. This allows comparing
+// parent TIDs that might be different threads of the same process.
+func NormalizeParentTid(p *Process) {
+	if p.Parent.Tid == 0 {
+		return
+	}
+
+	taskDir := fmt.Sprintf("/proc/%d/task", p.Parent.Tid)
+	entries, err := os.ReadDir(taskDir)
+	if err != nil {
+		// If we can't read the task dir, leave the TID as is
+		return
+	}
```

Comment:
I understand we cannot change the signature there due to calling context, but silencing error makes debugging harder.
Maybe only printing the error message could be a good middle ground.